worlds/sm64ex/Rules.py

```diff
@@ -1,67 +1,96 @@
     randomized_level_to_paintings = sm64_level_to_paintings.copy()
     randomized_level_to_secrets = sm64_level_to_secrets.copy()
-    if world.AreaRandomizer[player].value == 1:  # Some randomization is happening, randomize Courses
+    valid_move_randomizer_start_courses = [
+        "Bob-omb Battlefield", "Jolly Roger Bay", "Cool, Cool Mountain",
+        "Big Boo's Haunt", "Lethal Lava Land", "Shifting Sand Land",
+        "Dire, Dire Docks", "Snowman's Land"
+    ]  # Excluding WF, HMC, WDW, TTM, THI, TTC, and RR
+    if world.AreaRandomizer[player].value >= 1:  # Some randomization is happening, randomize Courses
         randomized_level_to_paintings = shuffle_dict_keys(world,sm64_level_to_paintings)
+        if world.AreaRandomizer[player].value < 3 and move_rando_bitvec > 0:
+            first_course = world.random.choice(valid_move_randomizer_start_courses)
+            original_entrance = next(entrance for entrance, painting in randomized_level_to_paintings.items() if painting == first_course)
+            randomized_level_to_paintings[original_entrance] = randomized_level_to_paintings[Entrances.BOB_OMB_BATTLEFIELD]
+            randomized_level_to_paintings[Entrances.BOB_OMB_BATTLEFIELD] = first_course
+
     if world.AreaRandomizer[player].value == 2:  # Randomize Secrets as well
         randomized_level_to_secrets = shuffle_dict_keys(world,sm64_level_to_secrets)
-    randomized_entrances = { **randomized_level_to_paintings, **randomized_level_to_secrets }
+    randomized_entrances = {**randomized_level_to_paintings, **randomized_level_to_secrets}
     if world.AreaRandomizer[player].value == 3:  # Randomize Courses and Secrets in one pool
-        randomized_entrances = shuffle_dict_keys(world,randomized_entrances)
+        randomized_entrances = shuffle_dict_keys(world, randomized_entrances)
         # Guarantee first entrance is a course
-        swapdict = { entrance: level for (level,entrance) in randomized_entrances }
-        if randomized_entrances[91] not in sm64_paintings_to_level.keys(): # Unlucky :C (91 -> BoB Entrance)
-            rand = world.random.choice(sm64_paintings_to_level.values())
-            randomized_entrances[91], randomized_entrances[swapdict[rand]] = rand, randomized_entrances[91]
-            swapdict[rand] = randomized_entrances[91]
-        swapdict.pop("Bob-omb Battlefield")
+        swapdict = {entrance: level for (level, entrance) in randomized_entrances.items()}
+        valid_first_courses = valid_move_randomizer_start_courses if move_rando_bitvec > 0 else list(sm64_paintings_to_level.keys())
+        first_course = randomized_entrances[Entrances.BOB_OMB_BATTLEFIELD]
+        if first_course not in valid_first_courses:  # Unlucky :C
+            rand = world.random.choice(valid_first_courses)
+            rand_entrance = swapdict[rand]
+            randomized_entrances[Entrances.BOB_OMB_BATTLEFIELD], randomized_entrances[rand_entrance] = rand, first_course
+            swapdict.pop(rand)
```

Comment:
Same as above.
Can probably be simplified to just use fix_reg as well.
`fix_reg(randomized_entrances, Entrances.BOB_OMB_BATTLEFIELD, <Courses not in valid_first_courses>, swapdict, world)`
You can then also remove the if cond just above, as its checked in fix_reg